docs/guide/start-identity-permissions-and-scopes.md

```diff
@@ -0,0 +1,201 @@
+of Alice. When Bob would authorize the FEA all calls with his bearer token by the FEA will set the
+`Yii::$app->user->identity` to Bob's identity.
+
+Since the identity is set, you can simply use `Yii::$app->user->can('role_you_want_to_check')` anywhere you want.
+E.g. `Yii::$app->user->can('admin')` would return `true` for Alice and `false` for Bob.
+
+> Note: Until now "scope" is never used and probably is not needed. 
+  Only when we have multiple clients, let's say FEA1 and FEA2, that need a different access scope for the *same* user,
+  scopes come into play.
+
+### Scope usage
+
+> Scopes only come into play in delegation scenarios, and always *limit* what a Client can do on *behalf* of a user:
+  a scope cannot allow an application to do more than what the user can do.
+
+Let's say (continuing from our sample above) that Bob (with the "user" role) wants to authorize two different clients:
+- Your own front-end application (FEA) to read, create and delete emails on his behalf.
+- A third-party app (from now on abbreviated as 3rdPA) to only create emails on his behalf.
+
+In that case you would need to define the scopes 'read_email', 'create_email' and 'delete_email' and assign them to the
+different Clients. Then Bob can specify which scopes the Client may use on his behalf during the authorization of the 
+Client.
+
+Note that another user, let's say Charles (who also has the role "user"), can allow different scopes for the same
+clients, e.g.:
+- FEA1 may read and create emails on his behalf.
+- 3rdPA may create and also delete emails on his behalf.
+
+When scopes are used you can use `Yii::$app->getModule('oauth2')->requestHasScope('scope_you_want_to_check')` to see if
+the client that makes the request has the required scope.
+
+So in the case of Bob and Charles:
+- `Yii::$app->user->can('admin')` would be `false` for both.
+- `Yii::$app->user->can('user')` would be `true` for both.
+- And when checking the scopes (`Yii::$app->getModule('oauth2')->requestHasScope('scope_you_want_to_check')`)
+  would yield the following (B = Bob, C = Charles):
+  |        |     'read_email'     |   'create_email'   |   'delete_email'    |
+  |--------|:--------------------:|:------------------:|:-------------------:|
+  | FEA1   |  B:`true`, C:`true`  | B:`true`, C:`true` | B:`true`, C:`false` |
+  | 3rdPA  | B:`false`, C:`false` | B:`true`, C:`true` | B:`false`, C:`true` |
+
+To summarize:
+- Roles define what the User may do.
+- Scopes *limit* what a Client may do on *behalf* of the User.
+- If you don't need to limit what a Client may do on behalf of a User, you probably don't need scopes.
+- If you do need scopes, you most likely need to check both the user Role and Scope. E.g.
+   ```php
+   class EmailController extends Controller
+   {
+       public function actionCreate()
+       {
+           if (
+               !Yii::$app->user->can('user')
+               || !Yii::$app->getModule('oauth2')->requestHasScope('create_email')
```

Comment:
@SOHELAHMED7 @mtangoo 
If I'm not mistaken they are now required both because of the `!`, in other words:
If the user does not have role 'user' or the request does not has the scope 'create_email' then throw the unauthorized exception.